src/stageros.cpp

```diff
@@ -92,24 +93,24 @@ class StageNode
         std::vector<Stg::ModelRanger *> lasermodels; //multiple rangers per position
 
         //ros publishers
-        ros::Publisher odom_pub; //one odom
-        ros::Publisher ground_truth_pub; //one ground truth
+        rclcpp::Publisher<nav_msgs::msg::Odometry>::SharedPtr odom_pub; //one odom
+        rclcpp::Publisher<nav_msgs::msg::Odometry>::SharedPtr ground_truth_pub; //one ground truth
 
-        std::vector<ros::Publisher> image_pubs; //multiple images
-        std::vector<ros::Publisher> depth_pubs; //multiple depths
-        std::vector<ros::Publisher> camera_pubs; //multiple cameras
-        std::vector<ros::Publisher> laser_pubs; //multiple lasers
+        std::vector<rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr> image_pubs; //multiple images
+        std::vector<rclcpp::Publisher<sensor_msgs::msg::Image>::SharedPtr> depth_pubs; //multiple depths
+        std::vector<rclcpp::Publisher<sensor_msgs::msg::CameraInfo>::SharedPtr> camera_pubs; //multiple cameras
+        std::vector<rclcpp::Publisher<sensor_msgs::msg::LaserScan>::SharedPtr> laser_pubs; //multiple lasers
 
-        ros::Subscriber cmdvel_sub; //one cmd_vel subscriber
+        rclcpp::Subscription<geometry_msgs::msg::Twist>::SharedPtr cmdvel_sub; //one cmd_vel subscriber
     };
 
     std::vector<StageRobot const *> robotmodels_;
 
     // Used to remember initial poses for soft reset
     std::vector<Stg::Pose> initial_poses;
-    ros::ServiceServer reset_srv_;
+    //TODO: ros::ServiceServer reset_srv_;
```

Comment:
I don't think we really use this, but i guess this means won't do :p

Reply:
We didn't use it, but for the sake of completeness I did implement it as a `std_srvs::srv::Trigger` service.